[Chrainx] iP

Chrainx opened the pull request:
# ***Chrainx*** **ChatBot** 👍
> Your Personal ChatBot

[Click Here](https://www.youtube.com/watch?v=dQw4w9WgXcQ&list=PL8dZXjD8meS_WZzEKSReIBPLzKaW3HboH&index=1) to see tutorial on how to use it.

Supported task:

- *Todo*
- *Deadline*
- *Event*


Available function:

- [x] Add
- [x] Delete
- [x] List
- [x] Mark
- [x] Unmark
- [x] Find
- [ ] Remind

Example:

```
1. [T][ ] Sleep
2. [D][ ] Homework (by: 17 Oct 2023)
3. [T][ ] Eat
```

sample code: 
```
   /**
     * Print introduction.
     */
    public void Hello() {
        printChat("Hello! I'm Chrainx\n" + "What can I do for you?" );
    }

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Chrainx committed:
level-0

Chrainx committed:
level-1

Chrainx committed:
level-2

Chrainx committed:
level-3

Chrainx committed:
level-4

Chrainx committed:
A-TextUiTesting

Chrainx committed:
level-6

Chrainx committed:
level-6

Chrainx committed:
level-7

Chrainx committed:
level-7

Chrainx committed:
level-7

Chrainx committed:
level-7

Chrainx committed:
Merge pull request #1 from Chrainx/branch-level-7

Branch level 7

Chrainx committed:
level-8

Chrainx committed:
Merge pull request #2 from Chrainx/branch-level-8

level-8

Chrainx committed:
A-MoreOOP

Chrainx committed:
Merge pull request #3 from Chrainx/A-MoreOOP

A-MoreOOP

Chrainx committed:
A-Packages

Chrainx committed:
Merge pull request #4 from Chrainx/A-Packages

A-Packages

Chrainx committed:
Merge pull request #5 from Chrainx/add-gradle-support

Add gradle support

Chrainx committed:
A-Gradle

Chrainx committed:
A-JUnit

Chrainx committed:
A-JavaDoc

Chrainx committed:
Level-9

Chrainx committed:
Merge branch-A-JavaDoc

Chrainx committed:
Merge branch-Level-9

saraozn commented on src/main/java/duke/Parser/Parser.java:
```diff
@@ -0,0 +1,114 @@
+                );
+                ui.addTask(targetTask, tasks.getNumberOfTask() + 1);
+                tasks.addTask(targetTask);
+            } else  {
```
```suggestion
            } else {
```

saraozn commented on src/main/java/duke/Parser/Parser.java:
```diff
@@ -0,0 +1,114 @@
+            } else {
+                throw new DukeException("This event is invalid");
+            }
+        } else if (command.equals("MARK")){
```
```suggestion
        } else if (command.equals("MARK")) {
```

saraozn commented on src/main/java/duke/Task/Deadlines.java:
```diff
@@ -0,0 +1,73 @@
+     * @throws DukeException Exception where the deadline is not valid.
+     */
+    public static boolean isDeadline(String input) throws DukeException {
+        if(input.split( " ")[0].equals("deadline")) {
```
```suggestion
        if (input.split( " ")[0].equals("deadline")) {
```

freddychenyouren2 replied:
```suggestion
        if (input.split(" ")[0].equals("deadline")) {
```

There seems to be an extra whitespace inside the `split()` method parameter.

saraozn commented on src/main/java/duke/Task/Deadlines.java:
```diff
@@ -0,0 +1,73 @@
+        if(input.split( " ")[0].equals("deadline")) {
+            if (input.split(" ").length == 1) {
+                throw new DukeException("OOPS! The description of deadline cannot be empty");
+            } else if (!input.contains("/by ")){
```
```suggestion
            } else if (!input.contains("/by ")) {
```

saraozn commented on src/main/java/duke/Task/Events.java:
```diff
@@ -0,0 +1,87 @@
+    public String writeString() {
+        if (this.getMarkStatus()) {
+            return "E,0," + this.getName() + "," + this.start + "," + this.end + "\n";
+        } else  {
```
```suggestion
        } else {
```

saraozn commented on src/main/java/duke/Task/Events.java:
```diff
@@ -0,0 +1,87 @@
+        if(input.split( " ")[0].equals("event")) {
+            if (input.split(" ").length == 1) {
+                throw new DukeException("OOPS! The description of event cannot be empty");
+            } else if (!input.contains("/from")){
```
```suggestion
            } else if (!input.contains("/from")) {
```

saraozn commented on src/main/java/duke/Task/Events.java:
```diff
@@ -0,0 +1,87 @@
+                throw new DukeException("OOPS! The description of event cannot be empty");
+            } else if (!input.contains("/from")){
+                throw new DukeException("OOPS! The description of event does not contain /from");
+            } else if (!input.contains("/to")){
```
```suggestion
            } else if (!input.contains("/to")) {
```

saraozn commented on src/main/java/duke/Task/Task.java:
```diff
@@ -0,0 +1,63 @@
+     * @throws DukeException Exception where the input is not a task.
+     */
+    public static boolean isTask(String input) throws DukeException {
+        if(input.split( " ")[0].equals("todo") || input.split( " ")[0].equals("deadline") || input.split( " ")[0].equals("event")) {
```
```suggestion
        if (input.split( " ")[0].equals("todo") || input.split( " ")[0].equals("deadline") || input.split( " ")[0].equals("event")) {
```
can place a line break to improve readability

saraozn commented on src/main/java/duke/Task/ToDos.java:
```diff
@@ -0,0 +1,50 @@
+    public String writeString() {
+        if (this.getMarkStatus()) {
+            return "T,0," + this.getName() + "\n";
+        } else  {
```
```suggestion
        } else {
```

saraozn commented on src/main/java/duke/Task/ToDos.java:
```diff
@@ -0,0 +1,50 @@
+     * @throws DukeException Exception where the todo is not valid.
+     */
+    public static boolean isTodo(String input) throws DukeException {
+        if(input.split( " ")[0].equals("todo")) {
```
```suggestion
        if (input.split( " ")[0].equals("todo")) {
```

saraozn commented on src/main/java/duke/Ui/Ui.java:
```diff
@@ -0,0 +1,111 @@
+ * Collection of the Ui
+ */
+public class Ui {
+    private final String Border = "____________________________________________________________";
```
```suggestion
    private final String BORDER = "____________________________________________________________";
```
constant names must be all uppercase

saraozn reviewed (commented):
Looks good except for a few minor coding standard violations. Good job!

freddychenyouren2 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+     */
+    private Scanner sc;
+    /**
+     * Variable to access the file where alll of the task is stored.
```
I spot a typo in the JavaDoc here! 😄 Did you mean "all"?

freddychenyouren2 commented on src/main/java/duke/DukeException/DukeException.java:
```diff
@@ -0,0 +1,23 @@
+    public String toString() {
+        return super.getMessage();
+    }
+}
```
Should the JavaDoc start with a verb-s instead of imperative? 

For instance, instead of `Create`, we write `Creates`. Instead of `Return`, we write `Returns`, and so on.

This issue is also present in the `@return` description.

You may view the exact guide [here](https://se-education.org/guides/conventions/java/intermediate.html#comments)

freddychenyouren2 commented on src/main/java/duke/Storage/Storage.java:
```diff
@@ -0,0 +1,93 @@
+     * @param tasks TaskList of Tasks created by the user.
+     * @throws DukeException Exception where the chatbot failed to edit the file.
+     */
+    public void editStorage (TaskList tasks) throws DukeException {
```
```suggestion
    public void editStorage(TaskList tasks) throws DukeException {
```
There is an extra whitespace between method name and method parameters. Be careful! 😄

freddychenyouren2 commented on src/main/java/duke/Task/Deadlines.java:
```diff
@@ -0,0 +1,73 @@
+    /**
+     * Deadline of the task.
+     */
+    private LocalDate ddl;
```
Might a more descriptive name like `deadline` be more suitable than `ddl`?

At first glance, `ddl` has less hints on what this field is about and how it might be used.

freddychenyouren2 commented on src/main/java/duke/Task/Events.java:
```diff
@@ -0,0 +1,87 @@
+                + " to: "
+                + this.end.format(DateTimeFormatter.ofPattern("MMM d yyyy"))
+                + ")";
+    }
```
Nice way of writing this `toString()` method! Makes it very readable 👍

freddychenyouren2 commented on src/main/java/duke/Task/Events.java:
```diff
@@ -0,0 +1,87 @@
+     * @throws DukeException Exception where the event is not valid.
+     */
+    public static boolean isEvent(String input) throws DukeException {
+        if(input.split( " ")[0].equals("event")) {
```
```suggestion
        if (input.split(" ")[0].equals("event")) {
```
Another place that has some whitespace issues.

freddychenyouren2 commented on src/main/java/duke/Task/ToDos.java:
```diff
@@ -0,0 +1,50 @@
+     * Create the todo task.
+     * @param name Description of the task.
+     */
+    public ToDos (String name) {
```
```suggestion
    public ToDos(String name) {
```

freddychenyouren2 commented on src/main/java/duke/Ui/Ui.java:
```diff
@@ -0,0 +1,111 @@
+            listOfTask = listOfTask + tasks.getTask(i).toString() + "\n";
+        }
+        printChat(listOfTask);
+    }
```
Perhaps you may want to maintain consistency of method names using verbs and avoid gerunds (verb + ing)

So in this case, `list(TaskList tasks)` may be better than `listing(TaskList tasks)`. An alternative name could be `listAllTasks(TaskList tasks)` to make it more descriptive and easier to read.

freddychenyouren2 reviewed (commented):
Overall good job in terms of implementation! Be careful on the small details that causes coding violations.

Also, I believe JavaDocs should start with verb-s words. (For example: `Returns`, `Creates`, `Prints`, `Converts`, and so on.)

There are also some minor issues in consistency with naming of methods.
For more information, you may also want to have a look at this Medium article on [Clean Code (Best practice for naming) Part 1](https://ahmedmahmoud-eltaher.medium.com/clean-code-best-practice-for-naming-part-1-f67ebe8c0894).

Chrainx committed:
level-10

Chrainx committed:
level-10

Chrainx committed:
Merge pull request #6 from Chrainx/branch-level-10

Branch level 10

Chrainx committed:
Mark, Unmark, Delete Cases: Handle Negative index inputed by user

Mark, Unmark, Delete Cases is not handling input with negative number

Add a base case to handle negative index

Chrainx committed:
A Assertions

Chrainx committed:
Some methods have worngly written java doc and indentation

To follow the coding standard

Add s(es) to verb and fix indentation

Chrainx committed:
Merge pull request #8 from Chrainx/branch-A-Assertions

Branch a assertions

Chrainx committed:
Merge pull request #9 from Chrainx/A-CodeQuality

Some methods have worngly written java doc and indentation

Chrainx committed:
newTask

Chrainx committed:
Merge pull request #10 from Chrainx/branch-A-CodeQuality

newTask

Chrainx committed:
Fix Bug

Chrainx committed:
BetterGUI

Chrainx committed:
solveBug

Chrainx committed:
addMeow

Chrainx committed:
fixSpacing

Chrainx committed:
adjustPeriod

Chrainx committed:
userGuide

Chrainx committed:
divideLongMethod

Chrainx committed:
jarRelease

Chrainx committed:
fixMeowMessages

Chrainx committed:
javaFxVersionUpdate

Chrainx committed:
addTestCase

Chrainx committed:
fixTestCase

Chrainx committed:
fixBugs

Chrainx committed:
fixByeBugs
